code/src/reducers/todos.js

```diff
@@ -0,0 +1,65 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  list: {
+    name: "Test list",
+    items: [],
+  },
+};
+
+export const todos = createSlice({
+  name: "todos",
+  initialState: initialState,
+  reducers: {
+    addTodo: (state, action) => {
+      const itemInfo = action.payload;
+      state.list.items.push(itemInfo);
+    },
+    setDone: (state, action) => {
+      const { itemIndex, done } = action.payload;
+      state.list.items[itemIndex].done = done;
+    },
+    removeTodo: (state, action) => {
+      const { itemIndex } = action.payload;
+      state.list.items = state.list.items.filter(
+        (item, index) => index !== itemIndex
+      );
+    },
+    removeAllTodos: (state) => {
+      state.list.items = [];
+    },
+    setAll: (state) => {
+      state.list.items.forEach((item) => {
+        item.done = true;
+      });
+    },
+    unSetAll: (state) => {
+      state.list.items.forEach((item) => {
+        item.done = false;
+      });
+    },
```

Comment:
maybe this could be one action which takes a payload of true or false and toggle accordingly?
This is totally fine, just a way to enhance a bit maybe.
```
 toggleAll: (state, action) => {
      state.list.items.forEach((item) => {
        item.done = action.payload; // either true or false
      });
    },
```